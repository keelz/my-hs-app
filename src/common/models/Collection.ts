import { ICard } from './Card';

export interface ICollection {
  cards: ICard[];
}
