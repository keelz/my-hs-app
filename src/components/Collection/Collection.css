.Collection {
  margin-top: 1.25rem;
}

.Collection img:hover {
  cursor: pointer;
}

.Collection-align-left {
  left: 0;
}

.Collection-align-right {
  right: 0;
}

.Collection-body {
  position: relative;
  margin-top: .625rem;
  padding-left: 5rem;
  padding-right: 5rem;
  display: flex;
  justify-content: center;
  flex-wrap: wrap;
  height: 34.375rem;
  overflow: hidden;
  border-radius: .3125rem;
  background-color: rgb(235, 235, 235);
}

.Collection-body-nav-button {
  position: absolute;
  top: 0;
  width: 3.75rem;
  height: 100%;
  background: transparent;
  font-size: 3rem;
}

.Collection-body-nav-button-icon {
  display: flex;
  flex-direction: column;
  justify-content: center;
  height: 100%;
  filter: opacity(.05);
  color: #000000;
}

.Collection-body-nav-button-icon.active {
  filter: opacity(1);
  cursor: pointer;
}

.Collection-card {
  height: 17.1875rem;
  padding: .3125rem;
}

.Collection-classbar {
  flex-grow: 1;
}

.Collection-footer {
  margin-top: 1.25rem;
}

.Collection-header {
  display: flex;
}

.Collection-mana-bar {
  padding: .625rem;
  background-color: rgb(235, 235, 235);
  border-radius: .3125rem;
  width: 18.75rem;
}

.Collection-play-style {
  margin-right: .625rem
}

.Collection-play-style img {
  height: 2.8125rem;
}
