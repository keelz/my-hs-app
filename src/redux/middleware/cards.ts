import { Dispatch } from 'redux';
import { AppMiddlewareApi, IActionType } from '../Types';
import cardsService, { ICardsService } from '../../common/services/cards.service';
import {
  CARDS_SET_ACTIVE_CLASSNAME,
  CARDS_SET_CARDS,
  CARDS_SET_FILTER,
  CARDS_SET_PAGINATION,
} from '../reducers/Cards';

export const composeMiddleware = (service: ICardsService) =>
  (api: AppMiddlewareApi) =>
  (next: Dispatch<IActionType<any>>) =>
  (action: IActionType<any>) => {
    switch (action.type) {
      case CARDS_SET_ACTIVE_CLASSNAME:
        return service.handleSetActiveClassname(api)(next)(action);
      case CARDS_SET_CARDS:
        next(action);
        service.resetPagination(api)(action);
        return;
      case CARDS_SET_FILTER:
        return service.handleSetFilter(api)(next)(action);
      case CARDS_SET_PAGINATION:
        return service.handleSetPagination(api)(next)(action);
      default:
        return next(action);
    }
  };

const middleware = composeMiddleware(cardsService);

export default middleware;
