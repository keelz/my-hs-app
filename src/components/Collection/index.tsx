import * as React from 'react';
import './Collection.css';
import { ICollection } from '../../common/models/Collection';
import { CardClassName } from '../../common/models/Card';
import CollectionBody from './Body';
import CollectionHeader from './Header';

export interface ICollectionStateProps {
  activeCardClassName: CardClassName;
  collection: ICollection;
}

const Collection: React.SFC<ICollectionStateProps> = props =>
  <div className="Collection">
    <CollectionHeader />
    <CollectionBody collection={props.collection} />
  </div>;

export default Collection;
