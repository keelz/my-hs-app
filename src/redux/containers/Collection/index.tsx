import { connect } from 'react-redux';
import { IRootState } from '../../Types';
import { CardClassName } from '../../../common/models/cards.model';
import {
  selectActiveCardClassName,
  selectCardsWithFilters
} from '../../selectors/cards';
import Collection from '../../../components/Collection';

const mapStateToProps = (state: IRootState) => ({
  activeCardClassName: selectActiveCardClassName(state) as CardClassName,
  collection: {
    cards: selectCardsWithFilters(state),
  },
});

const ConnectedCollection = connect(mapStateToProps)(Collection);

export default ConnectedCollection;
