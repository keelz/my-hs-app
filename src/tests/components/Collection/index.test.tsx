import * as React from 'react';
import * as enzyme from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Collection from '../../../components/Collection';
import { CardClassName, ICard } from '../../../common/models/cards.model';
import cards from '../../../common/mocks/collection';

const composeProps = () => ({
  activeCardClassName: CardClassName.DRUID,
  collection: {
    cards: [...cards].slice(0, 50) as ICard[],
  },
});

describe('Collection', () => {
  describe('snapshots', () => {
    it('renders without crashing', () => {
      const tree = enzyme.shallow(<Collection {...composeProps()} />);
      expect(shallowToJson(tree)).toMatchSnapshot();
    });
  });
});
