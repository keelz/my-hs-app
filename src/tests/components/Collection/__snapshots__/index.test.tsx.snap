// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Collection snapshots renders without crashing 1`] = `
<div
  className="Collection"
>
  <CollectionHeader />
  <Connect(CollectionBody)
    collection={
      Object {
        "cards": Array [
          Object {
            "artist": "Nutthapon Petchthai",
            "cardClass": "MAGE",
            "collectible": true,
            "cost": 5,
            "dbfId": 2539,
            "flavor": "It's on the rack next to ice lance, acid lance, and English muffin lance.",
            "id": "AT_001",
            "name": "Flame Lance",
            "playRequirements": Object {
              "REQ_MINION_TARGET": 0,
              "REQ_TARGET_TO_PLAY": 0,
            },
            "rarity": "COMMON",
            "set": "TGT",
            "text": "Deal $8 damage to a minion.",
            "type": "SPELL",
          },
          Object {
            "artist": "Tooth",
            "cardClass": "MAGE",
            "collectible": true,
            "cost": 3,
            "dbfId": 2541,
            "flavor": "Burning man, brah.",
            "id": "AT_002",
            "mechanics": Array [
              "SECRET",
            ],
            "name": "Effigy",
            "rarity": "RARE",
            "set": "TGT",
            "text": "<b>Secret:</b> When a friendly minion dies, summon a random minion with the same Cost.",
            "type": "SPELL",
          },
          Object {
            "artist": "Arthur Bozonnet",
            "attack": 3,
            "cardClass": "MAGE",
            "collectible": true,
            "cost": 2,
            "dbfId": 2545,
            "flavor": "And he can't get up.",
            "health": 2,
            "id": "AT_003",
            "mechanics": Array [
              "HEROPOWER_DAMAGE",
            ],
            "name": "Fallen Hero",
            "rarity": "RARE",
            "set": "TGT",
            "text": "Your Hero Power deals 1 extra damage.",
            "type": "MINION",
          },
          Object {
            "artist": "Gabor Szikszai",
            "cardClass": "MAGE",
            "collectible": true,
            "cost": 1,
            "dbfId": 2572,
            "flavor": "Now with 100% more blast!",
            "id": "AT_004",
            "mechanics": Array [
              "RECEIVES_DOUBLE_SPELLDAMAGE_BONUS",
            ],
            "name": "Arcane Blast",
            "playRequirements": Object {
              "REQ_MINION_TARGET": 0,
              "REQ_TARGET_TO_PLAY": 0,
            },
            "rarity": "EPIC",
            "referencedTags": Array [
              "SPELLPOWER",
            ],
            "set": "TGT",
            "text": "Deal $2 damage to a minion. This spell gets double bonus from <b>Spell Damage</b>.",
            "type": "SPELL",
          },
          Object {
            "artist": "Mike Sass",
            "cardClass": "MAGE",
            "collectible": true,
            "cost": 3,
            "dbfId": 2542,
            "flavor": "It's always Huffer.",
            "id": "AT_005",
            "name": "Polymorph: Boar",
            "playRequirements": Object {
              "REQ_MINION_TARGET": 0,
              "REQ_TARGET_TO_PLAY": 0,
            },
            "rarity": "RARE",
            "referencedTags": Array [
              "CHARGE",
            ],
            "set": "TGT",
            "text": "Transform a minion into a 4/2 Boar with <b>Charge</b>.",
            "type": "SPELL",
          },
          Object {
            "artist": "Dan Scott",
            "attack": 3,
            "cardClass": "MAGE",
            "collectible": true,
            "cost": 4,
            "dbfId": 2549,
            "flavor": "Is he aspiring or inspiring?  Make up your mind!",
            "health": 5,
            "id": "AT_006",
            "mechanics": Array [
              "INSPIRE",
            ],
            "name": "Dalaran Aspirant",
            "rarity": "COMMON",
            "referencedTags": Array [
              "SPELLPOWER",
            ],
            "set": "TGT",
            "text": "<b>Inspire:</b> Gain <b>Spell Damage +1</b>.",
            "type": "MINION",
          },
          Object {
            "artist": "Andrew Hou",
            "attack": 3,
            "cardClass": "MAGE",
            "collectible": true,
            "cost": 3,
            "dbfId": 2571,
            "flavor": "Does he sling spells, or do his spells linger about.  Who can say?",
            "health": 4,
            "id": "AT_007",
            "mechanics": Array [
              "BATTLECRY",
            ],
            "name": "Spellslinger",
            "rarity": "COMMON",
            "set": "TGT",
            "text": "<b>Battlecry:</b> Add a random spell to each player's hand.",
            "type": "MINION",
          },
          Object {
            "artist": "Christopher Moeller",
            "attack": 6,
            "cardClass": "MAGE",
            "collectible": true,
            "cost": 6,
            "dbfId": 2544,
            "flavor": "The Grand Tournament has a \\"No dragons allowed\\" policy, but it's rarely enforced.",
            "health": 6,
            "id": "AT_008",
            "name": "Coldarra Drake",
            "race": "DRAGON",
            "rarity": "EPIC",
            "set": "TGT",
            "text": "You can use your Hero Power any number of times.",
            "type": "MINION",
          },
          Object {
            "artist": "Zoltan & Gabor",
            "attack": 7,
            "cardClass": "MAGE",
            "collectible": true,
            "cost": 8,
            "dbfId": 2546,
            "elite": true,
            "flavor": "A masterless shamurai.",
            "health": 7,
            "id": "AT_009",
            "mechanics": Array [
              "DEATHRATTLE",
            ],
            "name": "Rhonin",
            "rarity": "LEGENDARY",
            "set": "TGT",
            "text": "<b>Deathrattle:</b> Add 3 copies of Arcane Missiles to your hand.",
            "type": "MINION",
          },
          Object {
            "artist": "Brandon Kitkouski",
            "attack": 3,
            "cardClass": "HUNTER",
            "collectible": true,
            "cost": 5,
            "dbfId": 2552,
            "flavor": "Not getting trampled is really the trick here.",
            "health": 3,
            "id": "AT_010",
            "mechanics": Array [
              "BATTLECRY",
            ],
            "name": "Ram Wrangler",
            "rarity": "RARE",
            "set": "TGT",
            "text": "<b>Battlecry:</b> If you have a Beast, summon a
random Beast.",
            "type": "MINION",
          },
          Object {
            "artist": "Alex Garner",
            "attack": 3,
            "cardClass": "PRIEST",
            "collectible": true,
            "cost": 4,
            "dbfId": 2555,
            "flavor": "She really likes seeing people get better.  That's why she hurts them in the first place.",
            "health": 5,
            "id": "AT_011",
            "mechanics": Array [
              "TRIGGER_VISUAL",
            ],
            "name": "Holy Champion",
            "rarity": "COMMON",
            "set": "TGT",
            "text": "Whenever a character is healed, gain +2 Attack.",
            "type": "MINION",
          },
          Object {
            "artist": "Dave Allsop",
            "attack": 5,
            "cardClass": "PRIEST",
            "collectible": true,
            "cost": 4,
            "dbfId": 2551,
            "flavor": "What did you expect to happen?  He's a Spawn.  Of Shadows.",
            "health": 4,
            "id": "AT_012",
            "mechanics": Array [
              "INSPIRE",
            ],
            "name": "Spawn of Shadows",
            "rarity": "RARE",
            "set": "TGT",
            "text": "<b>Inspire:</b> Deal 4 damage to each hero.",
            "type": "MINION",
          },
          Object {
            "artist": "Mike Sass",
            "cardClass": "PRIEST",
            "collectible": true,
            "cost": 1,
            "dbfId": 2568,
            "flavor": "The promise of glory is a powerful tool to get minions to do your bidding.  Only slightly less powerful than the promise of an ice cream bar!",
            "id": "AT_013",
            "name": "Power Word: Glory",
            "playRequirements": Object {
              "REQ_MINION_TARGET": 0,
              "REQ_TARGET_TO_PLAY": 0,
            },
            "rarity": "COMMON",
            "set": "TGT",
            "text": "Choose a minion. Whenever it attacks, restore 4 Health to
your hero.",
            "type": "SPELL",
          },
          Object {
            "artist": "Warren Mahy",
            "attack": 3,
            "cardClass": "PRIEST",
            "collectible": true,
            "cost": 3,
            "dbfId": 2566,
            "flavor": "Hopes to be promoted to \\"Shadowfriend\\" someday.",
            "health": 3,
            "id": "AT_014",
            "mechanics": Array [
              "TRIGGER_VISUAL",
            ],
            "name": "Shadowfiend",
            "rarity": "EPIC",
            "set": "TGT",
            "text": "Whenever you draw a card, reduce its Cost by (1).",
            "type": "MINION",
          },
          Object {
            "artist": "Dan Dos Santos",
            "cardClass": "PRIEST",
            "collectible": true,
            "cost": 2,
            "dbfId": 2563,
            "flavor": "\\"Are you interested in... HEALTH benefits?!\\"",
            "id": "AT_015",
            "name": "Convert",
            "playRequirements": Object {
              "REQ_ENEMY_TARGET": 0,
              "REQ_MINION_TARGET": 0,
              "REQ_TARGET_TO_PLAY": 0,
            },
            "rarity": "RARE",
            "set": "TGT",
            "text": "Put a copy of an enemy minion into your hand.",
            "type": "SPELL",
          },
          Object {
            "artist": "Sean O'Danield",
            "cardClass": "PRIEST",
            "collectible": true,
            "cost": 2,
            "dbfId": 2564,
            "flavor": "This minion is really powerful!",
            "id": "AT_016",
            "name": "Confuse",
            "rarity": "EPIC",
            "set": "TGT",
            "text": "Swap the Attack and Health of all minions.",
            "type": "SPELL",
          },
          Object {
            "artist": "Slawomir Maniak",
            "attack": 2,
            "cardClass": "NEUTRAL",
            "collectible": true,
            "cost": 4,
            "dbfId": 2569,
            "flavor": "A result of magical experiments carried out by the Black Dragonflight, it's not his fault that he's a vicious killer.",
            "health": 6,
            "id": "AT_017",
            "mechanics": Array [
              "BATTLECRY",
            ],
            "name": "Twilight Guardian",
            "race": "DRAGON",
            "rarity": "EPIC",
            "referencedTags": Array [
              "TAUNT",
            ],
            "set": "TGT",
            "text": "<b>Battlecry:</b> If you're holding a Dragon, gain +1 Attack and <b>Taunt</b>.",
            "type": "MINION",
          },
          Object {
            "artist": "Chris Rahn",
            "attack": 5,
            "cardClass": "PRIEST",
            "collectible": true,
            "cost": 7,
            "dbfId": 2556,
            "elite": true,
            "flavor": "She sees into your past and makes you face your fears.  Most common fear:  Getting Majordomo out of Sneed's Old Shredder.",
            "health": 4,
            "id": "AT_018",
            "mechanics": Array [
              "INSPIRE",
            ],
            "name": "Confessor Paletress",
            "rarity": "LEGENDARY",
            "set": "TGT",
            "text": "<b>Inspire:</b> Summon a random <b>Legendary</b> minion.",
            "type": "MINION",
          },
          Object {
            "artist": "Alex Horley Orlandelli",
            "attack": 1,
            "cardClass": "WARLOCK",
            "collectible": true,
            "cost": 4,
            "dbfId": 2822,
            "flavor": "Crescendo himself summoned this steed, riding it to victory in the Grand Tournament.  Wherever he rides, an army of riders ride behind him, supporting the legendary champion.",
            "health": 1,
            "id": "AT_019",
            "mechanics": Array [
              "DEATHRATTLE",
            ],
            "name": "Dreadsteed",
            "race": "DEMON",
            "rarity": "EPIC",
            "set": "TGT",
            "text": "<b>Deathrattle:</b> At the end
 of the turn, summon a Dreadsteed.",
            "type": "MINION",
          },
          Object {
            "artist": "Anton Zemskov",
            "attack": 6,
            "cardClass": "WARLOCK",
            "collectible": true,
            "cost": 7,
            "dbfId": 2624,
            "flavor": "They were originally called Cuddleguards, but they were not inspiring the proper amount of fear.",
            "health": 8,
            "id": "AT_020",
            "name": "Fearsome Doomguard",
            "race": "DEMON",
            "rarity": "COMMON",
            "set": "TGT",
            "type": "MINION",
          },
          Object {
            "artist": "Raymond Swanland",
            "attack": 3,
            "cardClass": "WARLOCK",
            "collectible": true,
            "cost": 2,
            "dbfId": 2777,
            "flavor": "\\"No, no, no. I asked for a tiny JESTER of evil.\\"",
            "health": 2,
            "id": "AT_021",
            "mechanics": Array [
              "TRIGGER_VISUAL",
            ],
            "name": "Tiny Knight of Evil",
            "race": "DEMON",
            "rarity": "RARE",
            "set": "TGT",
            "text": "Whenever you discard a card, gain +1/+1.",
            "type": "MINION",
          },
          Object {
            "artist": "Matt Dixon",
            "cardClass": "WARLOCK",
            "collectible": true,
            "cost": 4,
            "dbfId": 2628,
            "flavor": "* Not actually Jaraxxus' fist.",
            "id": "AT_022",
            "mechanics": Array [
              "InvisibleDeathrattle",
            ],
            "name": "Fist of Jaraxxus",
            "rarity": "RARE",
            "set": "TGT",
            "text": "When you play or discard this, deal $4 damage to a random enemy.",
            "type": "SPELL",
          },
          Object {
            "artist": "Skan Srisuwan",
            "attack": 5,
            "cardClass": "WARLOCK",
            "collectible": true,
            "cost": 6,
            "dbfId": 2537,
            "flavor": "We like to call him \\"Wesley\\".",
            "health": 4,
            "id": "AT_023",
            "mechanics": Array [
              "INSPIRE",
            ],
            "name": "Void Crusher",
            "race": "DEMON",
            "rarity": "RARE",
            "set": "TGT",
            "text": "<b>Inspire:</b> Destroy a random minion for each player.",
            "type": "MINION",
          },
          Object {
            "artist": "Kevin Chen",
            "cardClass": "WARLOCK",
            "collectible": true,
            "cost": 2,
            "dbfId": 2535,
            "flavor": "Very dangerous when attached to a demonbomb.",
            "id": "AT_024",
            "name": "Demonfuse",
            "playRequirements": Object {
              "REQ_MINION_TARGET": 0,
              "REQ_TARGET_TO_PLAY": 0,
              "REQ_TARGET_WITH_RACE": 15,
            },
            "rarity": "COMMON",
            "set": "TGT",
            "text": "Give a Demon +3/+3. Give your opponent a Mana Crystal.",
            "type": "SPELL",
          },
          Object {
            "artist": "Paul Mafayon",
            "cardClass": "WARLOCK",
            "collectible": true,
            "cost": 6,
            "dbfId": 2632,
            "flavor": "A prime example of lose-lose negotiating.",
            "id": "AT_025",
            "name": "Dark Bargain",
            "playRequirements": Object {
              "REQ_MINIMUM_ENEMY_MINIONS": 2,
            },
            "rarity": "EPIC",
            "set": "TGT",
            "text": "Destroy 2 random enemy minions. Discard 2 random cards.",
            "type": "SPELL",
          },
          Object {
            "artist": "Sojin Hwang",
            "attack": 4,
            "cardClass": "WARLOCK",
            "collectible": true,
            "cost": 2,
            "dbfId": 2623,
            "flavor": "After playing against 5 Annoy-O-Trons, any normal guard will become a Wrathguard.",
            "health": 3,
            "id": "AT_026",
            "mechanics": Array [
              "TRIGGER_VISUAL",
            ],
            "name": "Wrathguard",
            "race": "DEMON",
            "rarity": "COMMON",
            "set": "TGT",
            "text": "Whenever this minion takes damage, also deal that amount to your hero.",
            "type": "MINION",
          },
          Object {
            "artist": "Tooth",
            "attack": 4,
            "cardClass": "WARLOCK",
            "collectible": true,
            "cost": 6,
            "dbfId": 2621,
            "elite": true,
            "flavor": "He can summon anything, even a FEARSOME DOOMGUARD*.
*He's pretty sure this is going to work out.",
            "health": 4,
            "id": "AT_027",
            "name": "Wilfred Fizzlebang",
            "rarity": "LEGENDARY",
            "set": "TGT",
            "text": "Cards you draw from your Hero Power cost (0).",
            "type": "MINION",
          },
          Object {
            "artist": "Ryan Metcalf",
            "attack": 3,
            "cardClass": "ROGUE",
            "collectible": true,
            "cost": 5,
            "dbfId": 2765,
            "flavor": "He needed a break after that business in the Vale of Eternal Blossoms. Naturally, he chose to spend his vacation in an icy snowscape killing monsters.",
            "health": 7,
            "id": "AT_028",
            "mechanics": Array [
              "COMBO",
            ],
            "name": "Shado-Pan Rider",
            "rarity": "COMMON",
            "set": "TGT",
            "text": "<b>Combo:</b> Gain +3 Attack.",
            "type": "MINION",
          },
          Object {
            "artist": "Matt Dixon",
            "attack": 2,
            "cardClass": "ROGUE",
            "collectible": true,
            "cost": 1,
            "dbfId": 2715,
            "flavor": "The best part of buccaneering is the pants.",
            "health": 1,
            "id": "AT_029",
            "mechanics": Array [
              "TRIGGER_VISUAL",
            ],
            "name": "Buccaneer",
            "race": "PIRATE",
            "rarity": "COMMON",
            "set": "TGT",
            "text": "Whenever you equip a weapon, give it +1 Attack.",
            "type": "MINION",
          },
          Object {
            "artist": "Zoltan Boros",
            "attack": 3,
            "cardClass": "ROGUE",
            "collectible": true,
            "cost": 2,
            "dbfId": 2767,
            "flavor": "Almost went to play for Stormwind before signing with Undercity.",
            "health": 2,
            "id": "AT_030",
            "mechanics": Array [
              "COMBO",
            ],
            "name": "Undercity Valiant",
            "playRequirements": Object {
              "REQ_TARGET_FOR_COMBO": 0,
            },
            "rarity": "COMMON",
            "set": "TGT",
            "targetingArrowText": "Deal 1 damage.",
            "text": "<b>Combo:</b> Deal 1 damage.",
            "type": "MINION",
          },
          Object {
            "artist": "Alex Horley Orlandelli",
            "attack": 2,
            "cardClass": "ROGUE",
            "collectible": true,
            "cost": 2,
            "dbfId": 2766,
            "flavor": "He has a giant collection of purses now.  One for every outfit!",
            "health": 2,
            "id": "AT_031",
            "mechanics": Array [
              "TRIGGER_VISUAL",
            ],
            "name": "Cutpurse",
            "rarity": "RARE",
            "set": "TGT",
            "text": "Whenever this minion attacks a hero, add the Coin to your hand.",
            "type": "MINION",
          },
          Object {
            "artist": "Tooth",
            "attack": 4,
            "cardClass": "ROGUE",
            "collectible": true,
            "cost": 3,
            "dbfId": 2768,
            "flavor": "I have great deal for you... for 4 damage to your face!",
            "health": 3,
            "id": "AT_032",
            "mechanics": Array [
              "BATTLECRY",
            ],
            "name": "Shady Dealer",
            "rarity": "RARE",
            "set": "TGT",
            "text": "<b>Battlecry:</b> If you have a Pirate, gain +1/+1.",
            "type": "MINION",
          },
          Object {
            "artist": "Matt Dixon",
            "cardClass": "ROGUE",
            "collectible": true,
            "cost": 3,
            "dbfId": 2770,
            "flavor": "Yoink!",
            "id": "AT_033",
            "name": "Burgle",
            "rarity": "RARE",
            "set": "TGT",
            "text": "Add 2 random cards to your hand <i>(from your opponent's class)</i>.",
            "type": "SPELL",
          },
          Object {
            "artist": "Nutthapon Petchthai",
            "attack": 1,
            "cardClass": "ROGUE",
            "collectible": true,
            "cost": 4,
            "dbfId": 2763,
            "durability": 3,
            "flavor": "How much more poisoned can a blade get?  The answer is a lot.  A lot more poisoned.",
            "id": "AT_034",
            "name": "Poisoned Blade",
            "rarity": "EPIC",
            "set": "TGT",
            "text": "Your Hero Power gives this weapon +1 Attack instead of replacing it.",
            "type": "WEAPON",
          },
          Object {
            "artist": "Daren Bader",
            "cardClass": "ROGUE",
            "collectible": true,
            "cost": 3,
            "dbfId": 2587,
            "flavor": "Can you hold these eggs for just a second?  I promise they're not full of giant enraged undead spider things.",
            "id": "AT_035",
            "name": "Beneath the Grounds",
            "rarity": "EPIC",
            "set": "TGT",
            "text": "Shuffle 3 Ambushes into your opponent's deck. When drawn, you summon a 4/4 Nerubian.",
            "type": "SPELL",
          },
          Object {
            "artist": "Eric Braddock",
            "attack": 8,
            "cardClass": "ROGUE",
            "collectible": true,
            "cost": 9,
            "dbfId": 2586,
            "elite": true,
            "flavor": "Was actually a pretty nice guy before, you know, the whole Lich King thing.",
            "health": 4,
            "id": "AT_036",
            "mechanics": Array [
              "DEATHRATTLE",
            ],
            "name": "Anub'arak",
            "rarity": "LEGENDARY",
            "set": "TGT",
            "text": "<b>Deathrattle:</b> Return this to your hand and summon a 4/4 Nerubian.",
            "type": "MINION",
          },
          Object {
            "artist": "Dan Brereton",
            "cardClass": "DRUID",
            "collectible": true,
            "cost": 1,
            "dbfId": 2792,
            "flavor": "2 out of 2 saplings recommend that you summon the saplings.",
            "id": "AT_037",
            "mechanics": Array [
              "CHOOSE_ONE",
            ],
            "name": "Living Roots",
            "playRequirements": Object {
              "REQ_TARGET_TO_PLAY": 0,
            },
            "rarity": "COMMON",
            "set": "TGT",
            "text": "<b>Choose One -</b> Deal $2 damage; or Summon two 1/1 Saplings.",
            "type": "SPELL",
          },
          Object {
            "artist": "Laurel Austin",
            "attack": 2,
            "cardClass": "DRUID",
            "collectible": true,
            "cost": 2,
            "dbfId": 2782,
            "flavor": "She loves mana crystals, she hates mana crystals.   So fickle!",
            "health": 3,
            "id": "AT_038",
            "mechanics": Array [
              "BATTLECRY",
              "DEATHRATTLE",
            ],
            "name": "Darnassus Aspirant",
            "rarity": "RARE",
            "set": "TGT",
            "text": "<b>Battlecry:</b> Gain an empty Mana Crystal.
<b>Deathrattle:</b> Lose a Mana Crystal.",
            "type": "MINION",
          },
          Object {
            "artist": "Alex Pascenko",
            "attack": 5,
            "cardClass": "DRUID",
            "collectible": true,
            "cost": 4,
            "dbfId": 2780,
            "flavor": "Maybe if you whistle a tune it will soothe him.  Yeah...  Try that.",
            "health": 4,
            "id": "AT_039",
            "mechanics": Array [
              "INSPIRE",
            ],
            "name": "Savage Combatant",
            "race": "BEAST",
            "rarity": "RARE",
            "set": "TGT",
            "text": "<b>Inspire:</b> Give your hero
+2 Attack this turn.",
            "type": "MINION",
          },
          Object {
            "artist": "James Ryman",
            "attack": 4,
            "cardClass": "DRUID",
            "collectible": true,
            "cost": 4,
            "dbfId": 2786,
            "flavor": "She was born to be something.  She is just not quite sure what yet...",
            "health": 4,
            "id": "AT_040",
            "mechanics": Array [
              "BATTLECRY",
            ],
            "name": "Wildwalker",
            "playRequirements": Object {
              "REQ_FRIENDLY_TARGET": 0,
              "REQ_MINION_TARGET": 0,
              "REQ_TARGET_IF_AVAILABLE": 0,
              "REQ_TARGET_WITH_RACE": 20,
            },
            "rarity": "COMMON",
            "set": "TGT",
            "text": "<b>Battlecry:</b> Give a friendly Beast +3 Health.",
            "type": "MINION",
          },
          Object {
            "artist": "Ralph Horsley",
            "attack": 6,
            "cardClass": "DRUID",
            "collectible": true,
            "cost": 7,
            "dbfId": 2788,
            "flavor": "He gets a discount on the tournament entry fee because he is his own horse.",
            "health": 6,
            "id": "AT_041",
            "name": "Knight of the Wild",
            "rarity": "RARE",
            "set": "TGT",
            "text": "Whenever you summon a Beast, reduce the Cost of this card by (1).",
            "type": "MINION",
          },
          Object {
            "artist": "Arthur Gimaldinov",
            "attack": 2,
            "cardClass": "DRUID",
            "collectible": true,
            "cost": 2,
            "dbfId": 2783,
            "flavor": "That's saberTEETH, not like curved pirate blades.  That's a different kind of druid.  Druid of the Curved Pirate Blades.",
            "health": 1,
            "id": "AT_042",
            "mechanics": Array [
              "CHOOSE_ONE",
            ],
            "name": "Druid of the Saber",
            "rarity": "COMMON",
            "referencedTags": Array [
              "CHARGE",
              "STEALTH",
            ],
            "set": "TGT",
            "text": "[x]<b>Choose One -</b> Transform
into a 2/1 with <b>Charge</b>;
or a 3/2 with <b>Stealth</b>.",
            "type": "MINION",
          },
          Object {
            "artist": "Christopher Moeller",
            "cardClass": "DRUID",
            "collectible": true,
            "cost": 4,
            "dbfId": 2785,
            "flavor": "Hey!  Moon!  Can I have some mana crystals?",
            "id": "AT_043",
            "name": "Astral Communion",
            "rarity": "EPIC",
            "set": "TGT",
            "text": "Gain 10 Mana Crystals. Discard your hand.",
            "type": "SPELL",
          },
          Object {
            "artist": "Dan Scott",
            "cardClass": "DRUID",
            "collectible": true,
            "cost": 3,
            "dbfId": 2793,
            "flavor": "Is this a noun or a verb?  We will never know.",
            "id": "AT_044",
            "name": "Mulch",
            "playRequirements": Object {
              "REQ_MINION_TARGET": 0,
              "REQ_TARGET_TO_PLAY": 0,
            },
            "rarity": "EPIC",
            "set": "TGT",
            "text": "Destroy a minion.
Add a random minion to your opponent's hand.",
            "type": "SPELL",
          },
          Object {
            "artist": "Velvet Engine",
            "attack": 5,
            "cardClass": "DRUID",
            "collectible": true,
            "cost": 10,
            "dbfId": 2796,
            "elite": true,
            "flavor": "Call her \\"Tweety\\".  She'll find it real funny.  I PROMISE.",
            "health": 5,
            "id": "AT_045",
            "mechanics": Array [
              "AURA",
            ],
            "name": "Aviana",
            "rarity": "LEGENDARY",
            "set": "TGT",
            "text": "Your minions cost (1).",
            "type": "MINION",
          },
          Object {
            "artist": "Eva Widermann",
            "attack": 3,
            "cardClass": "SHAMAN",
            "collectible": true,
            "cost": 3,
            "dbfId": 2513,
            "flavor": "Turns out the tuskarr aren't real choosy about their totems.",
            "health": 2,
            "id": "AT_046",
            "mechanics": Array [
              "BATTLECRY",
            ],
            "name": "Tuskarr Totemic",
            "rarity": "COMMON",
            "set": "TGT",
            "text": "<b>Battlecry:</b> Summon a random basic Totem.",
            "type": "MINION",
          },
          Object {
            "artist": "RK Post",
            "attack": 4,
            "cardClass": "SHAMAN",
            "collectible": true,
            "cost": 4,
            "dbfId": 2613,
            "flavor": "It's nice to find a real craftsman in this day and age of mass-produced totems.",
            "health": 4,
            "id": "AT_047",
            "mechanics": Array [
              "BATTLECRY",
            ],
            "name": "Draenei Totemcarver",
            "rarity": "RARE",
            "set": "TGT",
            "text": "<b>Battlecry:</b> Gain +1/+1 for each friendly Totem.",
            "type": "MINION",
          },
          Object {
            "artist": "Luca Zontini",
            "cardClass": "SHAMAN",
            "collectible": true,
            "cost": 3,
            "dbfId": 2612,
            "flavor": "Favored by shaman who study the art of restoration and healing, this spell would feel smug, if it had feelings.",
            "id": "AT_048",
            "name": "Healing Wave",
            "playRequirements": Object {
              "REQ_TARGET_TO_PLAY": 0,
            },
            "rarity": "RARE",
            "set": "TGT",
            "text": "Restore #7 Health. Reveal a minion in each deck. If yours costs more, Restore #14 instead.",
            "type": "SPELL",
          },
          Object {
            "artist": "Sean McNally",
            "attack": 3,
            "cardClass": "SHAMAN",
            "collectible": true,
            "cost": 5,
            "dbfId": 2615,
            "flavor": "Allowing totems to attack is not cheating.  I mean, there isn't anything in the rule books about it.",
            "health": 6,
            "id": "AT_049",
            "mechanics": Array [
              "INSPIRE",
            ],
            "name": "Thunder Bluff Valiant",
            "rarity": "RARE",
            "set": "TGT",
            "text": "<b>Inspire:</b> Give your Totems +2 Attack.",
            "type": "MINION",
          },
        ],
      }
    }
  />
</div>
`;
